### PR TITLE
Add OpenAPI Specification for the /datafiles/ endpoint 

### LINKED ISSUE #2967: Add OpenAPI Specification for the /datafiles/ endpoint and fix the incorrect specs
## Problem
In order to ensure the accuracy of the specs generated by drf-spectacular for /datafiles/ endpoint , we will generate an OpenAPI Specification specifically for this endpoint. This will allow us to verify its correctness by comparing it with the existing test cases and the DRF browsable API page and also confirm it with the front-end team.

## Proposed solution
* Implement custom preprocessing hook function  to filter out all endpoints except for the /datafiles/ endpoint.The hook function selectively retains only the endpoint paths that match the /datafiles/ 
* Update the Spectacular settings in the  _common_settings.py_ file to include the custom preprocessing hook.
* Configure the PREPROCESSING_HOOKS setting with the custom hook function, we ensure that only the /datafiles/ endpoint is considered during the OpenAPI specification generation process.
* Finally, generate the spec file using the _./manage.py spectacular --color --file schema.yml_ command

### PR DESCRIPTION
**Pull request by varshadr**
Fixes #2967 

This PR adds OpenAPI Specification for the /datafiles/ endpoint to the codebase

**Technical details**
* Implemented custom preprocessing hook function  to filter out all endpoints except for the /datafiles/ endpoint.The hook function selectively retains only the endpoint paths that match the /datafiles/ 
* Updated the Spectacular settings in the  _common_settings.py_ file to include the custom preprocessing hook.
* Configured the PREPROCESSING_HOOKS setting with the custom hook function, we ensure that only the /datafiles/ endpoint is considered during the OpenAPI specification generation process.
* Finally, generated the spec file using the _./manage.py spectacular --color --file schema.yml_ command

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by silentninja — CHANGES_REQUESTED** [review]
This is a good start! Overall the spec looks good but there is some missing or incorrect information in the spec, please look at the specific comments below for more information.

**Comment by varshadr** [comment]
Thanks for the review @silentninja !
I'll consider the making the necessary changes

File: schema.yml [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+          application/x-www-form-urlencoded:
+            schema:
+              $ref: '#/components/schemas/PatchedDataFile'
+          multipart/form-data:
+            schema:
+              $ref: '#/components/schemas/PatchedDataFile'
+      security:
+      - tokenAuth: []
+      - cookieAuth: []
+      responses:
```
Comment (varshadr) [inline schema.yml:144]: @silentninja Do we expect here a 405 Method Not Allowed status code too?

**Comment by varshadr** [comment]
Closing this PR as some checks were not passed.

**Pull request closed without merging**